app/webpacker/controllers/column_preferences_controller.js

```diff
@@ -26,40 +26,17 @@ export default class ColumnPreferencesController extends Controller {
   #showHideColumn(e) {
     const element = e.target || e;
     const name = element.dataset.columnName;
-    const selector = `col[data-column-preferences-name="${name}"]`;
-    const column = this.table.querySelector(selector);
-    const index = this.#getIndex(column);
 
-    if (column == null) {
-      console.error(`ColumnPreferencesController: could not find ${selector}`);
-      return;
-    }
-
-    // Hide column definition
-    this.#showHideElement(column, element.checked);
-
-    // Hide each cell in column (ignore rows with colspan)
-    const rows = this.table.querySelectorAll("tr:not(:has(td[colspan]))");
-    rows.forEach((row) => {
-      // Ignore cell if spanning multiple columns
-      const cell = row.children[index];
-      if (cell == undefined) return;
-
-      this.#showHideElement(cell, element.checked);
-    });
+    this.table.classList.toggle(`hide-${name}`, !element.checked);
```

Comment:
Now the next step would be to have the column selector be labels for hidden checkboxes in the columns and couple the visibility of the columns with the value of the hidden input in CSS. Then you won't need JS at all. :wink:

Reply:
🤯 

That's so crazy it might just work

Reply:
I don't know why, but I tried it, and it worked with pretty minimal changes!
https://github.com/dacook/openfoodnetwork/commit/a46eded2321a7dd095d1e6767dca647d6b461aad

But pretty minimal benefit too, so I won't pursue it.

Reply:
Oh, I didn't know the `has` selector. That's clever. But as long as we need the column counter, we can't get rid of the JS. Hm, I would probably scrap the counter but I'm not the designer. :stuck_out_tongue_winking_eye:

Reply:
Haha, unfortunately the counter is needed in two places, where i've employed colspan to make rows span. One of them I think I could probably work around, but the other requires to span all visible rows.  Unfortunately there isn't a builtin solution for that. (you can use an arbitrary high number like `colspan="999"`, which automatically spans all columns, but this doesn't take into account hidden columns). 
Hmm, maybe it could be hacked from within the first column...

Reply:
I saw one example that used `visibility: collapse` which can hide entire columns. I haven't tested the effect on colspan though.